pkg/driver/memory/device_manager_memory.go

```diff
@@ -562,3 +563,40 @@ func (d *DeviceManager) GetUUID(u uuid.UUID) ([]byte, error) {
 	ur := &eveuuid.UuidResponse{Uuid: u.String()}
 	return proto.Marshal(ur)
 }
+
+func (d *DeviceManager) SetDeviceOptions(u uuid.UUID, b []byte) error {
+	dev, ok := d.devices[u]
+	if !ok {
+		return fmt.Errorf("no device UUID %s", u)
+	}
+	dev.Options = b
+	return nil
+}
+
+func (d *DeviceManager) GetDeviceOptions(u uuid.UUID) ([]byte, error) {
+	dev, ok := d.devices[u]
+	if !ok {
+		cfg := common.CreateBaseDeviceOptions(u)
+		err := d.SetDeviceOptions(u, cfg)
+		if err == nil {
+			return cfg, nil
+		}
+		return nil, fmt.Errorf("no device UUID %s", u)
+	}
+	return dev.Options, nil
+}
```

Comment:
Something about the logic here confuses me. Then function is `d.GetDeviceOptions()`, which is called to get the options set for a given device. If it finds the device, it returns `dev.Options`; makes sense.

If it doesn't find the _device_, it then sets the device options to a blank set, and then returns `dev.Options`? There are two issues here:

1. Couldn't `dev` actually be `nil`? Thus making `dev.Options` a segmentation fault?
2. If the device is not in the in-memory database, shouldn't we create it, rather than setting the device option?